BEP3.md

```diff
@@ -0,0 +1,156 @@
+
+1. there should be an easy way for users to swap unidirectionally or bidirectionally between Binance Chain and the other chain, better in a trustless way;
+2. there should be restrictions and/or transparency to ensure the total supply of tokens remained the same, and no one, even the issuer cannot freely change the circulation at will;
+3. there should be an easy way to calculate how many tokens are in circulation on each chain of both sides.
+
+Here new transaction and logics are required on Binance Chain, as it doesn’t support Smart Contract. Also a standard infrastructure and procedure should be proposed and used as best practice for such inter-chain communications.
+
+The BEP is to tackle the problem, and try to pave the broadway for a practical cross chain solution to empower the decentralized economy with Binance DEX.
+
+# Specification
+
+## Atomic Peg Swap
+
+The primary purpose of HTLC in Binance Chain is to support Atomic Peg Swap (APS). APS is proposed here as the standard way for asset/token issuers to peg or migrate part of the asset/token onto Binance Chain, so that users of the tokens can enjoy the fast transactions and pleasant trading experience.
+
+APS is designed to support peg token from any EVM based blockchain or any one with full smart contract features. Here in order to simplify the description, Ethereum is used as the most typical example.
+
+### Roles
+
+*   **Client**: users who want to swap tokens from Ethereum to Binance Chain, or the other way round;
+*   **Owner**: the issuers or anyone who want to peg the tokens from Ethereum to Binance Chain. In most cases, the Owner should be the issuer of the token. There expects one or multiple Owners to provide some service for one blockchain/project, depending on the size of the blockchain/project. 
+
+### Infrastructure Components
+
+*   **New transaction types on Binance Chain**: HTLT and CHLT transactions are used to lock and claim the asset to swap. The details are covered in the following section.
+*   **Swap smart contracts on Ethereum**: the APS contract are used to lock and claim the asset to swap too. The function should be similar to the new transaction types on Binance Chain. The details of interface covered in the following section.
+*   **Deputy process**: an application run and maintained by Owner to facilitate the automatic and continuous swap activity.
+*   **Client tooling**: wallets or other tools that help Clients to monitor the blockchains to complete the claim (and maybe the lock as well).
+
+### Protocol
+
+The protocol is the execution of 
+
+#### Preparation Stage
+
+1. Owner should issue proper number of tokens on Binance Chain as the Pegged 
+2. Owner should create one address on Binance Chain (as OB in the below diagram), and transfer in enough number of tokens for swap
```

Comment:
Is "`OB`" an ordinary address controlled by the owner's private key? If so, isn't the owner free to move tokens from this address?

Reply:
Yes. It is an ordinary address. But once HTLT transaction is sent, a certain number of coins will be transferred to a temporary address. There is no risk for swap.

Reply:
Understood the swap process works. My question is geared more towards ensuring the total supply of tokens. It's unclear to me how the current proposal guarantees at any given time, supply(BEP-2) + supply(ERC-20) == initialSupply(ERC-20)?

Reply:
Taking ERC20 token for instance, the total supply of token A  is 10000. If 5000 the BEP2 token is minted, then 5000 ERC20 token should be burned. Currently, there are 5000 BEP2 token and 5000 ERC20 token. Then the swap service provider use 1000 BEP2 token and 1000 ERC20 token for atomic swap service. Then total supply is still 10000.

Reply:
But there's no prevention against the token owner minting more BEP-2, besides community monitoring?

Reply:
You are right. The community monitor is the main insurance for the unexpected minting on Binance Chain.

Reply:
@HaoyangLiu I believe it cannot be called a peg then.

I will state again that I think it's needed to separate Atomic swap BEP and Peg BEP.